### PR TITLE
fix: Update default to not delete consumer groups

### PR DESCRIPTION
**Pull request by wcarlson5**
Signed-off-by: Walker Carlson <wcarlson@confluent.io>

### Description 
Don't delete consumer groups by default

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java [hunk 1]
```diff
@@ -240,7 +240,7 @@ public Optional<Materialization> getMaterialization(
 
   @Override
   public void stop() {
-    stop(false);
+    stop(true);
```
Comment (jnh5y) [inline ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: The variable name `isCreateOrReplace` on the various `stop()` methods is really, really confusing.

I'd suggest cleaning that up if possible.

Also, should this have a unit test?

Comment (wcarlson5) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: Yeah, sorry this isn't the full PR yet. Im not sure if changing the default is actually what we want to do. Defiantly change the shut down action. But with this change you should be able to test

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java [hunk 2]
```diff
@@ -405,7 +405,7 @@ public void onStateChange(final State newState, final State oldState) {
   @Override
   public void close() {
     loggerFactory.getLoggersWithPrefix(queryId.toString()).forEach(ProcessingLogger::close);
-    sharedKafkaStreamsRuntime.stop(queryId, false);
+    sharedKafkaStreamsRuntime.stop(queryId, true);
```
Comment (ableegoldman) [inline ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:408]: This naming feels a bit backwards to me, which I think is how we got into this mess in the first place actually -- to me, it sounds like `stop()` means to literally stop/kill the query, whereas `close()` feels like we're just temporarily closing it to clean up resources before we "open" it back up again.

Not saying we should change this in this PR, but at the very least let's circle back later with a PR to clarify which is which in the javadocs.

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java [hunk 3]
```diff
@@ -243,8 +243,8 @@ public void stop() {
     stop(false);
```
Comment (ableegoldman) [inline ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: One thing though -- can we make sure that we're still resetting the offsets if the query is actually explicitly terminated? I'm worried that we aren't doing that anymore.

 I think we should keep the default behavior of `stop` to mean "clean up the offsets" ie change this to true, but just make sure to pass in false when we're actually just stopping/pausing them in `QueryRegistryImpl` (ie [here](https://github.com/confluentinc/ksql/pull/9407/files#diff-3d6026517bf65fbc5c4619defec666b7a5261137ba02a69a66604137f84f47f3R461), as you're already doing)

Comment (ableegoldman) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: Another thing that's a bit too confusing as it is now but hopefully we can clean up in a later pass. Too many different meanings of `stop` and `close` floating around

Comment (wcarlson5) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: Yeah we have a unit test for that, and yeah I can keep the default to clean it up

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java (hunk 2 above)
Comment (wcarlson5) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:408]: Huh that makes sense to me, we stop like we would at a stop sign, we close all the resources on close

Comment (ableegoldman) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:408]: Fair enough -- I do think that the current way it is is the more popular/common meaning in this kind of code, it just hits a bit weird sometimes and I have to go through the code to figure out which is which. But either way docs are the answer

**Review by ableegoldman — APPROVED** [review]
LGTM, let's merge as soon as the tests pass (cc @rodesai )

File: ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java (hunk 1 above)
Comment (rodesai) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: We don't want to reset offsets here right?

Comment (rodesai) [reply ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java:243]: What actually calls this?

**Pull request merged**